### PR TITLE
[history] fix generated timeout wrapper

### PR DESCRIPTION
**Pull request by shijiesheng**
**What changed?**

https://github.com/uber/cadence/pull/5728 code generated history client should exclude some endpoints. 

**Why?**

Some replication endpoints don't have timeout before

**How did you test it?**

unit test

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: client/templates/timeout.tmpl [hunk 1]
```diff
@@ -3,6 +3,7 @@ import (
   "time"
 )
 
+{{$exclude := splitList "|" (index .Vars "exclude")}}
```
Comment (davidporter-id-au) [inline client/templates/timeout.tmpl:6]: I guess we're hitting the limitations of this templating

File: client/wrappers/timeout/history_generated_test.go [hunk 2]
```diff
@@ -107,3 +108,25 @@ func Test_historyClient_CloseShard(t *testing.T) {
 		})
 	}
 }
+
+func Test_historyClient_GetReplicationMessages(t *testing.T) {
+	t.Run("no timeout", func(t *testing.T) {
+		m := history.NewMockClient(gomock.NewController(t))
+		m.EXPECT().GetReplicationMessages(gomock.Any(), gomock.Any()).DoAndReturn(func(ctx context.Context, r *types.GetReplicationMessagesRequest, opt ...yarpc.CallOption) (*types.GetReplicationMessagesResponse, error) {
+			for {
```
Comment (davidporter-id-au) [inline client/wrappers/timeout/history_generated_test.go:116]: sorry, maybe just me being dumb, but I am not sure I understand why this is a loop?

File: client/wrappers/timeout/history_generated_test.go [hunk 3]
```diff
@@ -107,3 +108,25 @@ func Test_historyClient_CloseShard(t *testing.T) {
...
+				select {
+				case <-ctx.Done():
+					return nil, ctx.Err()
+				case <-time.After(time.Millisecond * 100):
```
[the remark this answers is not in the document]
Comment (shijiesheng) [reply client/wrappers/timeout/history_generated_test.go:120]: I tried. It's very complicated. The real issue here is incorrect test case with timeout set as 100. A quicker fix would be changing the timeout to 150ms and that should kill the flakiness

File: client/wrappers/timeout/history_generated_test.go (hunk 2 above)
Comment (shijiesheng) [reply client/wrappers/timeout/history_generated_test.go:116]: It's for wait and handle the context timeout.

File: client/templates/timeout.tmpl (hunk 1 above)
Comment (shijiesheng) [reply client/templates/timeout.tmpl:6]: Yes, but it's still useful to generate for most handlers.

Comment (davidporter-id-au) [reply client/templates/timeout.tmpl:6]: yeah, no concerns, just musing

File: client/wrappers/timeout/history_generated_test.go (hunk 2 above)
Comment (davidporter-id-au) [reply client/wrappers/timeout/history_generated_test.go:116]: It doesn't matter very much, so this is more a nit comment, but wouldn't jsut the select statement work?

Comment (shijiesheng) [reply client/wrappers/timeout/history_generated_test.go:116]: I think you are right. We don't have default case so it's not really needed.

File: client/wrappers/timeout/history_generated_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (taylanisikdemir) [reply client/wrappers/timeout/history_generated_test.go:120]: yeah let's go with higher number then

**Review by taylanisikdemir — APPROVED** [review]

**Pull request merged**